src/machines/appMachine.ts

```diff
@@ -0,0 +1,30 @@
+import { ActorRefFrom, createActor, setup } from 'xstate'
+import { authMachine } from './authMachine'
+import { useSelector } from '@xstate/react'
+import { ACTOR_IDS } from './machineConstants'
+
+const appMachine = setup({
+  actors: {
+    [ACTOR_IDS.AUTH]: authMachine,
+  },
+}).createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5gF8A0IB2B7CdGgAoBbAQwGMALASwzAEp8QAHLWKgFyqw0YA9EAjACZ0AT0FDkU5EA */
+  id: 'modeling-app',
+  invoke: [
+    {
+      src: ACTOR_IDS.AUTH,
+      systemId: ACTOR_IDS.AUTH,
+    },
+  ],
+})
+
+export const appActor = createActor(appMachine).start()
+
+export const authActor = appActor.system.get(ACTOR_IDS.AUTH) as ActorRefFrom<
+  typeof authMachine
+>
+export const useAuthState = () => useSelector(authActor, (state) => state)
+export const useToken = () =>
+  useSelector(authActor, (state) => state.context.token)
+export const useUser = () =>
+  useSelector(authActor, (state) => state.context.user)
```

Comment:
I'm personally not a fan of these very shallow uses of useSelector, but up to you :). It hides that useToken and useUser are just simple `useSelector` 's underneath.

Reply:
That XState comment is only talking about the selector _function_ but not `useSelector` itself (you see they're all _inside_ a React component).

I believe it makes sense to abstract the selector, but not the `useSelector` hook itself.

Reply:
Hm okay, so the developer has to know that they have to import two things to get the user or token?
```ts
import { useSelector } from 'xstate'
import { userSelector } from 'machines/appMachine'

// ...code code code...
const user = useSelector(userSelector)
```

Reply:
It'd be a common pattern to import useSelector; most files will already have it as we move forward with the migration to actors I'd think?

Reply:
Yeah okay that's fair, can do in a future PR